### PR TITLE
Fixed #65: restructured error handling.

### LINKED ISSUE #65: Stop exception propagation during dataflow
The eventually targets of a dataflow update should not affect whether or not they appear for the rest of the subscribers!

### PR DESCRIPTION
**Pull request by col6y**
Fixes #65.

This change is big; so don't approve it until you fully understand its consequences.

The old method of autodetachment for handling repeated errors was ugly and inelegant, and added lots of potential downstream bugs. This new system is much simpler and to-the-point: exceptions are propagated within the dataflow (but don't prevent other propagations) and caught at the dataflow entrypoints to be reported.

Also, the phidget subsystem got removed, because we decided it wasn't worth the maintenance burden.

### CODE REVIEW

File: CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public default void send(BooleanOutput output) {
     }
 
     public default EventOutput sendR(BooleanOutput output) {
-        output.set(get());
+        output.safeSet(get()); // TODO: should this be inconsistent like this?
```
Comment (amackworth) [inline CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java]: What do you mean by inconsistent?

File: CommonChickenRuntimeEngine/src/ccre/channel/BooleanStatus.java [hunk 2]
```diff
@@ -139,8 +140,21 @@ public final synchronized void set(boolean value) {
...
+                while (iterator.hasNext()) {
+                    EventOutput out2 = iterator.next();
+                    try {
+                        out2.event();
+                    } catch (Throwable ex) {
+                        e.addSuppressed(ex);
+                    }
+                }
+                throw e;
+            }
```
Comment (amackworth) [inline CommonChickenRuntimeEngine/src/ccre/channel/BooleanStatus.java]: The added complexity is a little annoying, but I suppose that's the price we pay for better safety. ¯\_(ツ)_/¯

**Comment by amackworth** [comment]
I really like this new approach to error handling with channels! Assuming it's been tested, it looks good to me.

Also, very glad we can finally :fire: the Phidget code!

**Comment by col6y** [comment]
It hasn't really been tested - there are no new test cases for the new functionality. I should go do that.

File: CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java (hunk 1 above)
Comment (col6y) [inline CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java]: send() uses set, but sendR() uses safeSet. I think I'll make them both use safeSet - this means that initial errors updating a channel will be caught, which may or may not be a good idea.

File: CommonChickenRuntimeEngine/src/ccre/channel/BooleanStatus.java (hunk 2 above)
Comment (col6y) [inline CommonChickenRuntimeEngine/src/ccre/channel/BooleanStatus.java]: Yeah. If you can think of a better way to write this piece of code, please let me know.

File: CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java (hunk 1 above)
Comment (amackworth) [inline CommonChickenRuntimeEngine/src/ccre/channel/BooleanInput.java]: Yeah, that makes sense to me.

**Comment by col6y** [comment]
@amackworth: I may have gone and done even more refactoring... at least now we only have one event dispatch implementation!

**Comment by col6y** [comment]
I'm merging this, even though it hasn't been thoroughly reviewed, because it ~needs~ to be merged. It's waited long enough, and I need to move on.

**Pull request merged**